### PR TITLE
docs: get-options documentation improved

### PR DESCRIPTION
**Pull request by retfah**
## Pull Request Checklist

- [X ] Have you added new tests to prevent regressions?
- [X ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [X ] Did you update the typescript typings accordingly (if applicable)?
- [X ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

The documentation of instance.get was lacking information when it simply returns instance.dataValues, or when it returns a copy of instance.dataValues. Searching a reason for this phenomenon, I also found the great 'clone'option, which was also lacking a description. Added both things.

### CODE REVIEW

**Review by fzn0x — COMMENTED** [review]
Maybe use `clone === true`

File: src/model.d.ts [hunk 1]
```diff
@@ -3390,13 +3390,14 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
...
   /**
-   * If no key is given, returns all values of the instance, also invoking virtual getters.
+   * If no key is given, returns all values of the instance, also invoking virtual getters. If an object has child objects or if clone===true, the object will be a copy. Otherwise, it will reference instance.dataValues.
    *
    * If key is given and a field or virtual getter is present for the key it will call that getter - else it
    * will return the value for key.
    *
    * @param options.plain If set to true, included instances will be returned as plain objects
    * @param options.raw  If set to true, field and virtual setters will be ignored
+   * @param options.clone Only evaluated when key==undefined. If set to true, all objects including child-objects will be copies and not reference the original dataValues-object. (default: false)
```
Comment (fzn0x) [inline src/model.d.ts:3400]: and this one key === undefined

**Review by fzn0x — DISMISSED** [review]

File: src/model.d.ts (hunk 1 above)
Comment (retfah) [reply src/model.d.ts]: I'm sorry. It is done now

**Comment by retfah** [comment]
I just realized that it would probably make more sense to reorder the types as well as the descriptions to be in alphabetical order. However, I have also seen that this seems not to be the case in the rest of the documentation. What is the offical sort order? By importance?

**Comment by ephys** [comment]
We don't have a sort order (it'd be really nice to have but we'd need to be able to enforce it with eslint or some other tool).

Now that I think of this, this part of the code is using legacy JSDoc instead of proper TSDoc. A better way to document its option bag would be to do this:

```ts
export interface ModelGetOptions {
  /** 
   * If set to true, included instances will be returned as plain objects
   */
  plain?: boolean;

  /**
   * Only evaluated when "key === undefined". If set to true, all objects including child-objects will be copies and not reference the original dataValues-object.
   * 
   * @default false
   */
  clone?: boolean;

  /**
   * If set to true, field and virtual setters will be ignored.
   */
  raw?: boolean;
}

class Model {
  /**
   * [rest of the method documentation]
   *
   * @param options Option bag. See {@link ModelGetOptions}.
   */
  get(options?: ModelGetOptions): TModelAttributes;
}
```

**Comment by retfah** [comment]
Please let me know whether I should implement your improvements with creating an interface or not. 
(It is my first contribution here and I think some more experienced contributors should decide this.)

**Comment by fzn0x** [comment]
> I just realized that it would probably make more sense to reorder the types as well as the descriptions to be in alphabetical order. However, I have also seen that this seems not to be the case in the rest of the documentation. What is the offical sort order? By importance?

We can do that in a separate PR IMO :)

**Comment by ephys** [comment]
Looks like I forgot to respond, but consider doing it in this PR. There isn't much point in merging this PR if we're going to replace it immediately afterwards by another solution

**Comment by retfah** [comment]
Here we go. I hope everything is right, since I'm not really used to typescript. Use it or not. 
Thank you @ephys for actually doing the work/creating the interface.

**Comment by fzn0x** [comment]
> Here we go. I hope everything is right, since I'm not really used to typescript. Use it or not. Thank you @ephys for actually doing the work/creating the interface.

The lint is not passed, can you fix it by any chance?

```
/home/runner/work/sequelize/sequelize/src/model.d.ts
Error:   2049:5  error  Trailing spaces not allowed         no-trailing-spaces
Error:   2051:6  error  Trailing spaces not allowed         no-trailing-spaces
Error:   2053:1  error  Trailing spaces not allowed         no-trailing-spaces
Error:   2054:6  error  Trailing spaces not allowed         no-trailing-spaces
Error:   2056:5  error  Trailing spaces not allowed         no-trailing-spaces
Error:   2063:5  error  Trailing spaces not allowed         no-trailing-spaces
Error:   34[15](https://github.com/sequelize/sequelize/actions/runs/3158428268/jobs/5140523255#step:6:16):1  error  More than 1 blank line not allowed  no-multiple-empty-lines
```

**Comment by retfah** [comment]
Lint fixed and merged with latest code. However, automatic check currently fails. Is the error really in the present pull request?

**Comment by WikiRik** [comment]
Seems like that was a flaky test, I've rerun that check

**Review by ephys — APPROVED** [review]

**Review by fzn0x — APPROVED** [review]

**Comment by WikiRik** [comment]
Thanks for working on this!

**Pull request merged**